app/root.tsx

```diff
@@ -9,9 +9,12 @@ import { getCurrentUser } from '~/modules/session/session.server';
 export const links: LinksFunction = () => [{ rel: 'stylesheet', href: stylesheet }];
 
 export async function loader({ request }: LoaderArgs) {
-  const currentUser = await getCurrentUser(request);
-
-  return json({ currentUser });
+  return json({
+    ENV: {
```

Comment:
We are loading PUBLIC environmental variables here that we want to expose to our client-side code. This pattern is suggested in the Remix docs [here](https://remix.run/docs/en/main/guides/envvars#browser-environment-variables).

Reply:
We could probably build a nice re-usable hook like `useEnv` to return the `ENV` object when needed!

```javascript
const env = useEnv();
console.log(env. PUBLIC_GOOGLE_CLIENT_ID);
```

Reply:
Great idea! I've created the `useEnv` hook and refactored the code to use it.